src/main/java/core/basesyntax/SquareSupplier.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class SquareSupplier {
+    private Random random = new Random();

```

Comment:
It's better to inject dependencies through the constructor. This way it's easier to control dependencies and it's much easier to test such code.

Reply:
I am not sure if I get you right. Should I remeved all Suppliers classes without figureSupplier and implement everything inside figureSupplier using line: new Square(size.getRandomGeometricValue(), color.getRandomColor()); ?

Reply:
I am not sure if I get you right. Should I remove all Suppliers classes without figureSupplier and implement everything inside figureSupplier using the line: new Square (size. getRandomGeometricValue(), color. getRandomColor()); ?